scmail.py

```diff
@@ -134,42 +143,61 @@ def register(ctx, fingerprint):
 
 
 @click.command()
-@click.option('--fingerprint', '-f', prompt='Enter fingerprint of mailbox', required=True, help='The fingerprint of the yourself mailbox.')
-@click.option('--sender-fingerprint', '-s', prompt='Enter the fingerprint of sender', default=None, help='The senders fingerprint.')
-@click.password_option('--password', '-p', prompt='Enter password of private key', help='The passphrase of private key')
+@click.option('--fingerprint', '-f', prompt='Enter fingerprint of mailbox',
+              required=True, type=str, help='The fingerprint of the yourself mailbox.')
+@click.option('--sender-fingerprint', '-s', prompt='Enter the fingerprint of sender',
+              type=str, help='The senders fingerprint.')
+@click.password_option('--password', '-p', prompt='Enter password of private key',
+                       help='The passphrase of private key')
 @click.pass_context
 def retrieve(ctx, fingerprint, sender_fingerprint, password):
     """Retrieve and Post messages from API"""
+    # Prepare send data.
     logging.debug(fingerprint, sender_fingerprint)
     data = {'fingerprint': fingerprint}
-    if not sender_fingerprint:
+    if sender_fingerprint == '':
         data.update(sender_fingerprint, sender_fingerprint)
-    # Retrieve
+
+    # Retrieve from api.
     r = requests.post(SECUREMAILBOX_URL + '/retrieve/', json=data)
     res = r.json()
-    logging.debug(f'response is: {res}')
 
     if r.status_code == 200:
         logging.info('The message retrieve successful.')
     else:
         logging.error(f'The message retrieve fail.\nError {r.status_code} is: {res.get("error")}')
         return
 
+    # load messages.
+    logging.debug(f'response is: {res}')
+    all_messages = res.get('messages')
```

Comment:
`all_messages` always contains `None` here.

Since our dictionary looks like the following:
```
{
  "data": {
    "count": 8, 
    "messages": [
      {
        "created_at": "Mon, 13 Apr 2020 22:41:46 GMT", 
        "message": "This is a test", 
        "sender_fingerprint": "8908218A320019CC0F37C80630C9054FDFA961B1", 
        "updated_at": "Mon, 13 Apr 2020 22:41:46 GMT"
      }, 
      {
        "created_at": "Wed, 15 Apr 2020 22:45:19 GMT", 
        "message": "Testing testing", 
        "sender_fingerprint": "ABF0C658E71F0AE90CB1519E7B2BBD4A6ED77E07", 
        "updated_at": "Wed, 15 Apr 2020 22:45:19 GMT"
      }
    ]
  }, 
  "error": null, 
  "success": true
}
```

We would need something like this:
```suggestion
    all_messages = res.get('data').get('messages')
```